src/pages/index.js

```diff
@@ -64,55 +80,176 @@ const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
 // Close image popup preview
 cardPreviewPopup.close();
 
+/* -------------------------------------------------------------------------- */
+/*                                Delete Popup                                */
+/* -------------------------------------------------------------------------- */
+
+// Create the delete popup instance
+const deletePopup = new Popup({ popupSelector: selectors.deletePopup });
+
+deletePopup.setEventListeners();
+
 /* -------------------------------------------------------------------------- */
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
 
 // This function creates a new card
-function createCard(data) {
-    const cardElement = new Card({ data, handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
-
-    }}, selectors.cardTemplate);
+function createCard(data, userId) {
+    const cardElement = new Card({ 
+        data,
+        handleImageClick: (imageData) => {
 
+            // Open image popup on click
+            cardPreviewPopup.open(imageData);
+        },
+        handleDeleteClick: () => {
+
+            // Open confirmation popup on click
+            deletePopup.open();
+
+            // Handle card deletion
+            api.deleteCard(data._id)
+            .then(() => {
+
+                // Render loading status
+                setSubmitButtonText(deleteConfirmButton, 'Deleting...');
```

Comment:
You should start the loading before api request

Reply:
Yep, I've moved that to the line above api request now.